climpred/tests/test_utils.py

```diff
@@ -177,3 +183,53 @@ def test_copy_coords_from_to_da_different_xro(control_ds_3d):
     with pytest.raises(ValueError) as excinfo:
         copy_coords_from_to(xro.isel(time=2).tos, c_1time)
     assert 'xro_from and xro_to must be both either' in str(excinfo.value)
+
+
+def test_cftime_index_unchanged():
+    """Tests that a CFTime index going through convert time is unchanged."""
+    inits = xr.cftime_range('1990', '2000', freq='Y', calendar='noleap')
+    da = xr.DataArray(np.random.rand(len(inits)), dims='init', coords=[inits])
+    new_inits = convert_time_index(da, 'init', '')
+    assert_allclose(new_inits.init, da.init)
+
+
+def test_pandas_datetime_converted_to_cftime():
+    """Tests that a pd.DatetimeIndex is converted to xr.CFTimeIndex."""
+    inits = pd.date_range('1990', '2000', freq='YS')
+    da = xr.DataArray(np.random.rand(len(inits)), dims='init', coords=[inits])
+    new_inits = convert_time_index(da, 'init', '')
+    assert isinstance(new_inits['init'].to_index(), xr.CFTimeIndex)
+
+
+def test_int64_converted_to_cftime():
+    """Tests the xr.Int64Index is converted to xr.CFTimeIndex."""
+    inits = np.arange(1990, 2000)
+    da = xr.DataArray(np.random.rand(len(inits)), dims='init', coords=[inits])
+    new_inits = convert_time_index(da, 'init', '')
+    assert isinstance(new_inits['init'].to_index(), xr.CFTimeIndex)
+
+
+def test_float64_converted_to_cftime():
+    """Tests the xr.Float64Index is converted to xr.CFTimeIndex."""
+    inits = np.arange(1990, 2000) * 1.0
+    da = xr.DataArray(np.random.rand(len(inits)), dims='init', coords=[inits])
+    new_inits = convert_time_index(da, 'init', '')
+    assert isinstance(new_inits['init'].to_index(), xr.CFTimeIndex)
+
+
+def test_convert_time_index_does_not_overwrite():
+    """Tests that `convert_time_index` does not overwrite the original index."""
+    inits = np.arange(1990, 2000)
+    da = xr.DataArray(np.random.rand(len(inits)), dims='init', coords=[inits])
+    new_inits = convert_time_index(da, 'init', '')
+    assert isinstance(da.init.to_index(), pd.Int64Index)
+    assert isinstance(new_inits.init.to_index(), xr.CFTimeIndex)
+
+
+def test_irregular_initialization_dates():

```

Comment:
Does this mean that DCPP CMIP5 Style units from only 1965, 1970, ... wouldn’t work?

Reply:
No they all work. That's what I was testing here -- that you can have a disjoint set of inits, e.g. `[1965, 1970, 1971']` and so on and it converts correctly. I.e. you don't need a full set `[1965, 1966, 1967, ..., 1971]`. Does that answer your question @aaronspring ?